From 0f7871693264712cf4b01e4965971d95a45a71de Mon Sep 17 00:00:00 2001
From: Ricard Wanderlof <polluxsynth@butoba.net>
Date: Mon, 14 Feb 2022 00:17:40 +0100
Subject: [PATCH 1845/1936] MiddleWare.cpp: Replace old raiseUi call with
 d.reply (#331)

All /alert messages are sent using d.reply(), so do the same when
/bank/save_to_slot fails, rather than using raiseUi().

Otherwise, if /bank/save_to_slot fails, the fusion UI will never
get the alert message.
---
 src/Misc/MiddleWare.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 4759aadd..f790462e 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -1540,10 +1540,8 @@ static rtosc::Ports middwareSnoopPortsWithoutNonRtParams = {
         impl.doReadOnlyOp([&impl,slot,part_id,&err](){
                 err = impl.master->bank.savetoslot(slot, impl.master->part[part_id]);});
         if(err) {
-            char buffer[1024];
-            rtosc_message(buffer, 1024, "/alert", "s",
+            d.reply("/alert", "s",
                     "Failed To Save To Bank Slot, please check file permissions");
-            GUI::raiseUi(impl.ui, buffer);
         }
         else d.broadcast("/damage", "s", "/bank/search_results/");
         rEnd},
-- 
2.47.0

